Dockerfile

```diff
@@ -1,100 +1,79 @@
 # This dockerfile expects proxies to be set via --build-arg if needed
 # It also expects to be contained in the /navigation2 root folder for file copy
 # Example build command:
-# sudo docker build -t nav2:latest --build-arg http_proxy=http://my.proxy.com:### .
-FROM osrf/ros2:bouncy-desktop
-
-SHELL ["/bin/bash", "-c"]
+# export http_proxy=http://my.proxy.com:80
+# export CMAKE_BUILD_TYPE=Debug
+# docker build -t nav2:latest --build-arg http_proxy --build-arg CMAKE_BUILD_TYPE ./
+FROM osrf/ros2:nightly
 
 # setup keys
-# check if proxy is set and get keys, using proxy if it is set
-RUN if [ "$http_proxy" == "" ]; \
+ARG http_proxy
+RUN if [ "$http_proxy" != "" ]; \
     then \
-      apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 \
-      --recv-keys 421C365BD9FF1F717815A3895523BAEEB01FA116; \
-    else \
       apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 \
       --keyserver-options http-proxy=$http_proxy \
       --recv-keys 421C365BD9FF1F717815A3895523BAEEB01FA116; \
     fi
 
-ENV ROS1_DISTRO melodic
-ENV ROS2_DISTRO bouncy
-ENV ROSDISTRO_INDEX_URL 'https://raw.githubusercontent.com/ros/rosdistro/master/index.yaml'
-
-# update latest package versions
-RUN apt-get update
-RUN rm -rf /etc/ros/rosdep/sources.list.d/20-default.list
-
 # install ROS2 dependencies
-RUN apt install -y \
-    build-essential \
-    cmake \
-    git \
-    python3-colcon-common-extensions \
-    python3-pip \
-    python-rosdep \
-    python3-vcstool \
-    wget
+RUN apt-get update && apt-get install -q -y \
+      build-essential \
+      cmake \
+      git \
+      python3-colcon-common-extensions \
+      python3-vcstool \
+      wget \
+    && rm -rf /var/lib/apt/lists/*
 
-# install Fast-RTPS dependencies
-RUN apt install --no-install-recommends -y \
-    libasio-dev \
-    libtinyxml2-dev
+# copy ros package repo
+ENV NAV2_WS /opt/nav2_ws
```

Comment:
Why move the nav2 workspace under /opt ? Is there a reason for that?

Reply:
Not much other than to keep image file heigheracy tidy. This might be helpful to organize the workspace that the entrypoint is sourcing, rather than polluting the root directory. This mainly just me trying to establish a template other ROS2 projects could replicate, or build upon. Say, perhaps if this navigation2 source build image is somehow reused or built `FROM` by another CI in ros-planning.